games/openrct2/Portfile

```diff
@@ -18,8 +19,37 @@ long_description    ${description} A construction and management simulation \
                     video game that simulates amusement park management. Requires \
                     files from the original RollerCoaster Tycoon 2 in order to work.
 
-checksums           rmd160  16fe40e2a07034f8571b4020634b23f10506f39a \
-                    sha256  51dd38d028bb352904162f63d1a4093d6580a702d8a4fc54eb98c3466b05ac69
+extract.suffix      .zip
+use_zip             yes
+
+set main_distfile       ${distfiles}
+
+set objects_distfile    objects${extract.suffix}
+set objects_mastersite  https://github.com/${github.author}/objects/releases/download/v1.0.18
+
+set ts_distfile         title-sequences${extract.suffix}
+set ts_mastersite       https://github.com/${github.author}/title-sequences/releases/download/v0.1.2c
```

Comment:
Is it legal for us to distribute these datafiles? What is their license -- is it the GPL-3+ as specified in the portfile? Are they the original copyrighted RCT files or new free files developed by OpenRCT?

I found this issue asking for clarification on the license of the title-sequence, with no decision reached: https://github.com/OpenRCT2/title-sequences/issues/3 though it does claim that the files are new free OpenRCT files. But I found no information about the license of the objects. The objects repo's readme says:

> Contains objects from RollerCoaster Tycoon 1, RollerCoaster Tycoon 2 and their expansion packs

That makes it sound like they are the original datafiles copyrighted by the original game's developers; if so, I wonder if we should distribute them.

If we are allowed to include the datafiles but their license is not GPL-3+ it may be beneficial to have them in a separate port so that the license can be specified for each port more clearly. This may be of benefit from other standpoints as well: maybe the code gets updated more often than the datafiles, so it would save space and time not to have to redownload the unchanged datafiles each time the code is updated. This was the reason why I implemented this separation in the alienarena and alienarena-data ports for example.

The port prints a note that reads:

> OpenRCT2 requires files from the original RollerCoaster Tycoon 2 in order to work

With the addition of these datafiles, is that still accurate?

Reply:
Regarding the object files, those aren't object files from the game. The original game is still required. The object files from openrct2 are simply metadata files that are used to load the original. IIRC these are JSON files that contain things like titles, localisations, etc.

If we don't add the other distfiles this way, the project will just download them during the configure phase (unless that is disabled).

Without doing this step, it becomes very difficult for a user to fix their installation to actually work, because these metadata files are required.

Reply:
Yup, one of the reasons we created our JSON objects to make fixing the metadata and localisation easier for the originals. JSON objects _can_ include images, but those for the original objects do not - they point to the appropriate DAT file, which OpenRCT2 then reads from the RollerCoaster Tycoon 2 install that the user supplied.

Reply:
@Gymnasiast what is the license of the title sequences and object images?

Reply:
We are working that out at the moment. A Debian maintainer raised exactly the same issue as you a few days ago.

Reply:
After discussing it, we have added CC-BY-4.0 licences to both title sequences and objects.

Reply:
Updated license.